src/java.base/share/classes/jdk/internal/foreign/abi/AbstractLinker.java

```diff
@@ -125,4 +134,30 @@ private static void checkHasNaturalAlignment(MemoryLayout layout) {
             throw new IllegalArgumentException("Layout bit alignment must be natural alignment: " + layout);
         }
     }
+
+    private static MemoryLayout stripNames(MemoryLayout ml) {
+        // we don't care about transferring alignment and byte order here
+        // since the linker already restricts those such that they will always be the same
+        return switch (ml) {
+            case StructLayout sl -> MemoryLayout.structLayout(stripNames(sl.memberLayouts()));
```

Comment:
shouldn't names be also stripped from enclosing struct/union/sequence?

Reply:
That should already happen because we're recreating the layout here.